### DISCUSSION TITLE
Using Vips as a cosmopolitan ape format ?

### DISCUSSION THREAD

**Discussion by SerJaimeLannister** [libvips/libvips#4281] [Ideas]
There is a project known as (https://github.com/jart/cosmopolitan)  which can convert itself to actually portable executable (https://justine.lol/ape.html) after which they can run cross platform (the same executable) on windows , mac , linux and the bsd's.

There are applications built on top of this https://redbean.dev/ & https://github.com/ahgamut/superconfigure which contains many languages like python , janet etc. and also many shell commands (if I remember correctly)

I recently tried out lua-libvips as a challenge to https://github.com/florinpop17/app-ideas
as the flip image challenge seemed interesting to try out in lua (https://github.com/florinpop17/app-ideas/blob/master/Projects/1-Beginner/Flip-Image-App.md)

but I was so frustrated by experience of installing vips on my system and also (the example in lua-vips had the use of thumbnail so maybe I needed openslide installed as well) 
Luckily I was on archlinux , so everything is in aur etc. but I could only just sympathesize with the linux users who aren't having these in their repo and they have to build it manually 

and you needed luarocks in the first place (which also required setting up $(eval(luarocks init)) or something along these lines.

But since lua already exists as a cosmopolitan project and luarocks etc. are rather a skill issue from my side. the only think that came to my mind to atleast make the burden a little less was rather having libvips be cross platform using cosmopolitan .

Since libvips is written in C , I think that it can be rather a lot easier to create an actually portable executable .

I am not exactly sure , cosmopolitan is one of the coolest projects I've ever seen , and I have just today seen libvips and I am also impressed insanely by libvips & how libvips have so many language bindings which feels very nice.

So I just wanted to present this idea to the community.

**Comment by jcupitt** [comment]
Hello @SerJaimeLannister,

I see they have build files for many popular libraries:

https://github.com/ahgamut/superconfigure/tree/main/lib

Someone could probably add libvips to that list pretty easily, although it seems to be very geared towards autotools. I don't think any of those packages use meson. It might need more research.
